tests/unit/Entity/EntityIdValueTest.php

```diff
@@ -186,4 +243,58 @@ public function invalidArrayProvider() {
 		];
 	}
 
+	/**
+	 * @param string $string
+	 *
+	 * @return EntityId
+	 */
+	private function newCustomId( $string ) {
+		return new EntityIdValueTest_CustomEntityId( $string );
+	}
+
 }
+
+class EntityIdValueTest_CustomEntityId extends EntityId {
+
+	/**
+	 * @param string $idSerialization
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct( $idSerialization ) {
```

Comment:
I believe you can remove this entire constructor and all tests will still succeed just fine.